container_wait.go

```diff
@@ -0,0 +1,108 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+var containerWaitCommand = &cli.Command{
+	Name:         "wait",
+	Usage:        "Block until one or more containers stop, then print their exit codes.",
+	ArgsUsage:    "CONTAINER [CONTAINER, ...]",
+	Action:       containerWaitAction,
+	BashComplete: waitBashComplete,
+}
+
+func containerWaitAction(clicontext *cli.Context) error {
+	if clicontext.NArg() == 0 {
+		return errors.Errorf("requires at least 1 argument")
+	}
+
+	client, ctx, cancel, err := newClient(clicontext)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+
+	walker := &containerwalker.ContainerWalker{
+		Client: client,
+		OnFound: func(ctx context.Context, found containerwalker.Found) error {
+			if err := waitContainer(clicontext, ctx, client, found.Container.ID()); err != nil {
```

Comment:
`nerdctl wait foo bar` may not be able to print the status of `bar` when `bar` exits and gets removed before `foo` exits.
So `task.Wait()` should be called in goroutines.

Reply:
Done